### PR TITLE
refactor: avoid casting to RemoteWebElement

### PR DESCRIPTION
**Pull request by asolntsev**
* instead, cast to interface HasIdentity (because it's a weaker type).
* this commit is similar to https://github.com/appium/java-client/pull/432/files

Some background: as a developer of framework which uses proxies for WebElement (Selenide), I cannot create proxy for RemoteWebElement since it's an abstract class, not interface.

## Change list

Please provide briefly described change list which are you going to propose. 
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] No changes in production code.
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

Please provide more details about changes if it is necessary. If there are new features you can provide code samples which show the way they
work and possible use cases. Also you can create [gists](https://gist.github.com) with pasted java code samples or put them here using markdown. 
About markdown please read [Mastering markdown](https://guides.github.com/features/mastering-markdown/) and [Writing on GitHub](https://help.github.com/categories/writing-on-github/)

### CODE REVIEW

File: src/main/java/io/appium/java_client/touch/offset/ElementOption.java [hunk 1]
```diff
@@ -85,8 +85,8 @@ public ElementOption withElement(WebElement element) {
         checkNotNull(element);
         checkArgument(true, "Element should be an instance of the class which "
                 + "extends org.openqa.selenium.remote.RemoteWebElement",
-                (RemoteWebElement.class.isAssignableFrom(element.getClass())));
-        elementId = RemoteWebElement.class.cast(element).getId();
+                (HasIdentity.class.isAssignableFrom(element.getClass())));
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/touch/offset/ElementOption.java:88]: why not to use instanceof?

File: src/main/java/io/appium/java_client/touch/offset/ElementOption.java [hunk 2]
```diff
@@ -85,8 +85,8 @@ public ElementOption withElement(WebElement element) {
...
+        elementId = HasIdentity.class.cast(element).getId();
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/touch/offset/ElementOption.java:89]: can explicit typecast be used here?

File: src/main/java/io/appium/java_client/touch/offset/ElementOption.java [hunk 3]
```diff
@@ -85,8 +85,8 @@ public ElementOption withElement(WebElement element) {
         checkNotNull(element);
         checkArgument(true, "Element should be an instance of the class which "
                 + "extends org.openqa.selenium.remote.RemoteWebElement",
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/touch/offset/ElementOption.java:87]: this message should be updated

File: src/main/java/io/appium/java_client/touch/offset/ElementOption.java (hunk 1 above)
Comment (asolntsev) [reply src/main/java/io/appium/java_client/touch/offset/ElementOption.java:88]: I don't know. I just tried to make a minimal change. You never know what can go wrong. :)

Comment (mykola-mokhnach) [reply src/main/java/io/appium/java_client/touch/offset/ElementOption.java:88]: Nothing is ideal. Lets do it properly

File: src/main/java/io/appium/java_client/touch/offset/ElementOption.java (hunk 3 above)
Comment (asolntsev) [reply src/main/java/io/appium/java_client/touch/offset/ElementOption.java]: done

File: src/main/java/io/appium/java_client/touch/offset/ElementOption.java (hunk 2 above)
Comment (asolntsev) [reply src/main/java/io/appium/java_client/touch/offset/ElementOption.java]: done

File: src/main/java/io/appium/java_client/touch/offset/ElementOption.java (hunk 1 above)
Comment (asolntsev) [reply src/main/java/io/appium/java_client/touch/offset/ElementOption.java]: done

**Pull request merged**